src/Guide/Types/Core.hs

```diff
@@ -131,6 +132,15 @@ instance A.ToJSON ItemKind where
   toJSON Other = A.object [
     "tag"      A..= ("Other" :: Text) ]
 
+instance A.FromJSON ItemKind where
+  parseJSON ik@(A.Object o) = do
+    iki <- A.parseJSON ik
```

Comment:
This instance loops indefinitely. `parseJSON ik = do iki <- parseJSON ik`

You should extract the tag from the object, then branch on it.

Also, please use `withObject` instead of inspecting the value manually, it gives nicer error messages (you get told what type the instance was parsing when it failed).